src/AbstractWrapperCachingEventManager.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace Dhii\WpEvents;
+
+/**
+ * Common functionality for normalized WordPress event managers that cache
+ * handler wrappers.
+ *
+ * @since [*next-version*]
+ */
+abstract class AbstractWrapperCachingEventManager extends AbstractNormalizedEventManager
+{
+    /**
+     * A cache of callback wrappers, by hash.
+     *
+     * @since [*next-version*]
+     *
+     * @var callable[]
+     */
+    protected $callbackWrappers = array();
+
+    /**
+     * {@inheritdoc}
+     *
+     * For the same name/callback pair, returns the same wrapper.
+     *
+     * @since [*next-version*]
+     */
+    protected function _getHandlerWrapper($name, $callback)
+    {
+        $cbHash = $this->_hashEventHandler($name, $callback);
+        if (!isset($this->callbackWrappers[$cbHash])) {
+            $this->callbackWrappers[$cbHash] = $this->_createHandlerWrapper($name, $callback);
+        }
+
+        return $this->callbackWrappers[$cbHash];
+    }
+
+    /**
+     * Hashes an event name / event handler pair.
+     *
+     * @since [*next-version*]
+     *
+     * @param string   $name    The name of the event.
+     * @param callable $handler The handler of the event.
+     *
+     * @return string The hash of the pair.
+     */
+    protected function _hashEventHandler($name, $handler)
+    {
+        $handlerHash = $this->_hashCallable($handler);
+        $pair        = $name . '|' . $handlerHash;
+
+        return $this->_hashScalar($pair);
+    }
+
+    /**
+     * Computes a hash of a given callable.
+     *
+     * @since [*next-version*]
+     *
+     * @param callable $callable The callable to hash.
+     *
+     * @throws \InvalidArgumentException If not a valid callable.
+     *
+     * @return string A hash of the callable.
+     */
+    protected function _hashCallable($callable)
+    {
+        if (\is_object($callable)) {
+            return $this->_hashObject($callable);
+        }
+
+        if (\is_array($callable)) {
+            return $this->_hashArray($callable);
+        }
+
+        throw new \InvalidArgumentException('Could not hash: not a valid callback');
+    }
+
+    /**
+     * Computes a hash of the array.
+     *
+     * Accounts for nested arrays.
+     *
+     * @since [*next-version*]
+     *
+     * @param array $array The array to hash.
+     *
+     * @return string A hash of the array.
+     */
+    protected function _hashArray(array $array)
+    {
+        $itemHashes = array();
+        foreach ($array as $_idx => $_item) {
+            if (\is_array($_item)) {
```

Comment:
Consider using [`gettype()`][1] and a `switch` statement with type class constants for switch cases.

[1]: http://php.net/manual/en/function.gettype.php